Automattic/woocommerce-services
# Load the FeatureAnnouncement modal to the order listing page

## Description
This PR makes the migration admin notice's "Confirm update" button functional.
![image](https://github.com/Automattic/woocommerce-services/assets/572862/0a5880a9-a046-4075-b14d-7171528628cb)

Upon clicking the button, the migration will install, and activate WCShipping and WCTax. 
![image](https://github.com/Automattic/woocommerce-services/assets/572862/0dc05686-99f5-425a-a1bd-a6f7ed2c6c64)

This is the same as clicking the "Update now" directly in the modal, introduced in this PR https://github.com/Automattic/woocommerce-services/pull/2745.

## Technical decision
### [main.js](https://github.com/Automattic/woocommerce-services/blob/trunk/client/main.js#L54) vs wcshipping-migration-admin-notice.js
To load `<FeatureAnnouncement>` on the order list page, we could use the existing `getRouteClass` pattern found in [main.js](https://github.com/Automattic/woocommerce-services/blob/trunk/client/main.js#L54). This way, we wouldn't need to create our own webpack's entry point. In order to use this pattern, we need to follow the `enqueue_wc_connect_script()`  code to load the javascript and style, etc. There are also a lot of variables added in here that we don't need. For the sake of the migration modal, I thought it would be simpler (and we'd have more control) to create our own [entry point](https://github.com/Automattic/woocommerce-services/pull/2756/files#diff-1fb26bc12ac780c7ad7325730ed09fc4c2c3d757c276c3dacc44bfe20faf166fR58) and render the `<FeatureAnnouncement>` component inside the PHP admin notice banner. This way, we also have more control over the modal without worrying about breaking other pages. The downside is loading the `isEligable` props which is from the redux store. It turned out it wasn't too bad because we only needed the [shipping label store](https://github.com/Automattic/woocommerce-services/pull/2756/files#diff-cafa06d18d10a101b583ffbe68a81c29f97cacbf981e3e79167354c18786efa2R21). As a result, I chose to stick with `client/wcshipping-migration-admin-notice.js` instead of following the `main.js` pattern.

### Clicking "Confirm update" to trigger React `<FeatureAnnouncement>` button action
Instead of creating our own React component for the admin notice. I continued to use the WP's one from the PHP's side. To trigger the migration inside the PHP's admin notice, I added an event listener to render the `<FeatureAnnouncement>` component with the store context. Then, the script continues and will click the "Update now" button to trigger the migration. 

### Related issue(s)
NA

### Steps to test
1. Go to `woocommerce-services/classes/class-wc-connect-service-settings-store.php` and replace the following `is_eligible_for_migration` function. This function decides if the modal should popup or not
```
		public function is_eligible_for_migration() {
			WC_Connect_Options::delete_option( 'wcshipping_migration_state' );
			$migration_state = WC_Connect_Options::get_option( 'wcshipping_migration_state' );
			//TODO: Return $migration_state check && connect-server flag
			return WC_Connect_WCST_To_WCShipping_Migration_State_Enum::COMPLETED !== $migration_state && true;
		}
```
2. Make sure you have "WooCommerce Shipping" and "WooCommerce Tax" in your plugins folder. We can't test the "install" part because they aren't part of dotorg yet.
3. Go to the order listing page.
4. Press F12 and check the Network tab, click the "Confirm update" button in the admin notice banner ![image]
(https://github.com/Automattic/woocommerce-services/assets/572862/0a5880a9-a046-4075-b14d-7171528628cb)
5. Confirm the 3 requests did go through and confirm the modal popped up and the button in the modal reads "Updating...".
![image](https://github.com/Automattic/woocommerce-services/assets/572862/ab43438a-2cd9-491d-999d-968e44f914a9)
6. Confirm you are being redirected to the plugins page with WCS&T deactivated, WCShipping and WCTax activated.

### Checklist
